packages/driver/src/cy/multi-domain/util.ts

```diff
@@ -0,0 +1,24 @@
+import _ from 'lodash'
+import { AssertionError } from 'chai'
+
+export const correctStackForCrossDomainError = (serializedError: any, userInvocationStack: string) => {
+  //  Since Errors sent over postMessage need to be serialized to Objects, we need to serialize them back into Error instances
+  let errorClass
+
+  switch (serializedError?.name) {
+    case 'AssertionError':
+      errorClass = AssertionError
+      break
+    default:
+      errorClass = Error
+  }
+
+  // only assign missing properties on the object to reify the error properly
+  const reifiedError = _.assignWith(new errorClass(serializedError?.message), serializedError, (objValue, srcValue) => {
```

Comment:
```suggestion
  const ErrorClass = serializedError?.name === 'AssertionError' ? AssertionError : Error

  // only assign missing properties on the object to reify the error properly
  const reifiedError = _.assignWith(new ErrorClass(serializedError?.message), serializedError, (objValue, srcValue) => {
```

Reply:
yeah this switch statement is a bit overkill 😂